### ISSUE TITLE
Cogen instances for Function

### ISSUE THREAD

**Issue by xuwei-k** [rickynils/scalacheck#190]

**Comment by rickynils** [comment]
@xuwei-k I'm not sure what issue you're describing here. Could you provide some more details?

**Comment by xuwei-k** [comment]
https://github.com/scalaz/scalaz/pull/1095

For example, `scalaz.StoreT` is Comonad. Comonad has `cobindRightIdentity` law. 

https://github.com/scalaz/scalaz/blob/v7.2.0/core/src/main/scala/scalaz/Comonad.scala#L21

**Comment by xuwei-k** [comment]
```scala
F[A] => B // second parameter of cobindRightIdentity[A, B](fa: F[A], f: F[A] => B)`
```

if `A = Int`, `B = Int` ↓

```scala
F[Int] => Int
```

if `type F[a] = Store[Int, a]`  ↓

```scala
Store[Int, a] => Int
```

`Store[Int, a]` is identical to `Tuple2[Int => a, Int]` ↓


```scala
Tuple2[Int => a, Int] => Int
```


so I need `Cogen[Int => a]` when I want to `Tuple2[Int => a, Int] => Int`.



FYI: 
- scalaprops https://github.com/scalaprops/scalaprops/blob/v0.3.1/gen/src/main/scala/scalaprops/Cogen.scala#L32-L36
- quickcheck https://github.com/nick8325/quickcheck/blob/2.8.2/Test/QuickCheck/Arbitrary.hs#L314-L315

**Comment by non** [comment]
This was fixed in #259.

**Comment by xuwei-k** [comment]
> This was fixed in #259.

Thanks! 😃
